### PR TITLE
Move no-channels edge case logic from UI to server

### PR DESCRIPTION
**Pull request by dlipeles**
Improves the flow for users in the case where they have:

`Not added any channels but created a default banner`
`Then have added their first channels`

### CODE REVIEW

File: app/controllers/publishers_controller.rb [hunk 1]
```diff
@@ -423,7 +423,7 @@ def balance
   end
 
   def get_site_banner_data
-    create_site_banners()
+    prepare_site_banner_data()
```
Comment (corymcdonald) [inline app/controllers/publishers_controller.rb]: no parenthesis on methods

File: app/controllers/publishers_controller.rb [hunk 2]
```diff
@@ -451,6 +451,9 @@ def create_site_banners
         channel.site_banner = SiteBanner.new_helper(current_publisher.id, channel.id)
       end
     end
+    if current_publisher.channels.length.zero?
+      current_publisher.update(default_site_banner_mode: true)
```
Comment (corymcdonald) [inline app/controllers/publishers_controller.rb]: let's do this 
```ruby
current_publisher.update(default_site_banner_mode: true) if current_publisher.channels.length.zero?
```

Comment (dlipeles) [reply app/controllers/publishers_controller.rb]: how bow dis? 

```
    if current_publisher.channels.length.zero?
      current_publisher.update(default_site_banner_mode: true)
    else
      current_publisher.channels.each do |channel|
        channel.site_banner = SiteBanner.new_helper(current_publisher.id, channel.id) if channel.site_banner.nil?
      end
    end
```

Comment (corymcdonald) [reply app/controllers/publishers_controller.rb]: sorry! I didn't see that it had an else statement. 👍

**Pull request merged**